src/invariant/LedgerEntryIsValid.cpp

```diff
@@ -641,15 +640,48 @@ LedgerEntryIsValid::checkIsValid(ConfigSettingEntry const& cfg,
         }
         break;
     case ConfigSettingID::CONFIG_SETTING_CONTRACT_EXECUTION_LANES:
-        if (cfg.contractExecutionLanes().ledgerMaxTxCount < 0)
+        break;
+    case ConfigSettingID::CONFIG_SETTING_CONTRACT_BANDWIDTH_V0:
+        if (cfg.contractBandwidth().feePropagateData1KB <= 0 ||
+            cfg.contractBandwidth().ledgerMaxPropagateSizeBytes <= 0 ||
+            cfg.contractBandwidth().txMaxSizeBytes <= 0)
         {
-            return "Invalid ledgerMaxTxCount";
+            return "Invalid contractBandwidth";
         }
         break;
-    case ConfigSettingID::CONFIG_SETTING_CONTRACT_BANDWIDTH_V0:
     case ConfigSettingID::CONFIG_SETTING_CONTRACT_COMPUTE_V0:
+        if (cfg.contractCompute().feeRatePerInstructionsIncrement <= 0 ||
+            cfg.contractCompute().ledgerMaxInstructions <= 0 ||
+            cfg.contractCompute().txMaxInstructions <= 0 ||
+            cfg.contractCompute().txMemoryLimit <= 0)
+        {
+            return "Invalid contractCompute";
+        }
+        break;
     case ConfigSettingID::CONFIG_SETTING_CONTRACT_HISTORICAL_DATA_V0:
+        if (cfg.contractHistoricalData().feeHistorical1KB <= 0)
+        {
+            return "Invalid feeHistorical1KB";
+        }
+        break;
     case ConfigSettingID::CONFIG_SETTING_CONTRACT_LEDGER_COST_V0:
+        if (cfg.contractLedgerCost().ledgerMaxReadLedgerEntries <= 0 ||
+            cfg.contractLedgerCost().ledgerMaxReadBytes <= 0 ||
+            cfg.contractLedgerCost().ledgerMaxWriteLedgerEntries <= 0 ||
+            cfg.contractLedgerCost().ledgerMaxWriteBytes <= 0 ||
+            cfg.contractLedgerCost().txMaxReadLedgerEntries <= 0 ||
+            cfg.contractLedgerCost().txMaxReadBytes <= 0 ||
+            cfg.contractLedgerCost().txMaxWriteLedgerEntries <= 0 ||
+            cfg.contractLedgerCost().txMaxWriteBytes <= 0 ||
+            cfg.contractLedgerCost().feeReadLedgerEntry <= 0 ||
+            cfg.contractLedgerCost().feeWriteLedgerEntry <= 0 ||
+            cfg.contractLedgerCost().feeRead1KB <= 0 ||
+            cfg.contractLedgerCost().bucketListTargetSizeBytes <= 0 ||
+            cfg.contractLedgerCost().writeFee1KBBucketListLow <= 0 ||
+            cfg.contractLedgerCost().writeFee1KBBucketListHigh <= 0 ||
+            cfg.contractLedgerCost().bucketListWriteFeeGrowthFactor > 0)
```

Comment:
This looks incorrect (should be <0?)

Reply:
Yeah I committed the invariant change prematurely. Fixing now.